roryhofland/vue3-treeselect
# Fix behavior of Left, Right, Home and End keys, Update Dependencies

### Related Item(s)
[#2245](https://github.com/ramp4-pcar4/ramp4-pcar4/issues/2245)

### Changes
- Removed code that handled behavior of the Right, Left, Home and End keys
- Updated dependencies (by running `npm update`)
- Changed the hashing algorithm used during building from md4 to md5

### Testing
Testing would have to be done using a local RAMP repo

1. Within `package.json`, change the line `"@ramp4-pcar4/vue3-treeselect": "github:ramp4-pcar4/vue3-treeselect"` to `"@ramp4-pcar4/vue3-treeselect": "github:IshavSohal/vue3-treeselect",`
2. Run `npm install`
3. Run the following to clear any cache 
```
cd node_modules
rm -r .vite
cd ..
npm cache clear --force
```
4. Run `npm run dev` to run an instance of RAMP
5. In the RAMP instance, press F12 to open devtools
6. Open any sample with a wizard (ex. Sample 27)
7. Click on the `+` in the legend to open the wizard
8. Add https://section917.canadacentral.cloudapp.azure.com/arcgis/rest/services/TestData/EcoAction/MapServer as a Map Image Layer 
9. Click though the wizard to reach Step 3, where the tree picker appears
10. Type `natu` 
11. Click the Left arrow key, and observe that the cursor moves to the left on the first click
12. Click the Right arrow key, and observe that the cursor moves to the right on the first click
13. Click the Home key, and observe that the cursor moves to the very beginning of the search term
14. Click the End key, and observe that the cursor moves to the very end of the search term
15. Click the Down/Up arrow keys, and observe that the currently selected option is highlighted in gray 
16. Throughout this process, observe that there are no errors or warnings in the devtool console
